cmd/root.go

```diff
@@ -115,48 +99,93 @@ func initConfig() {
 	if err != nil {
 		ExitWithError(err.Error())
 	}
-
-	if err := networkConfig.ReadInConfig(); err == nil {
-		miners := networkConfig.GetStringSlice("miners")
-		sharders := networkConfig.GetStringSlice("sharders")
-		if len(miners) > 0 && len(sharders) > 0 {
-			zcncore.SetNetwork(miners, sharders)
-		}
-	}
 }
 
-func checkWalletPath(cmd *cobra.Command, args []string) {
+func loadConfigs() {
+	cfgConfig = viper.New()
+	cfgNetwork = viper.New()
 	var configDir string
 	if cDir != "" {
 		configDir = cDir
 	} else {
 		configDir = getConfigDir()
 	}
 
+	// ~/.zcn/config.yaml
+	cfgConfig.AddConfigPath(configDir)
+	if &cfgFile != nil && len(cfgFile) > 0 {
+		cfgConfig.SetConfigFile(configDir + "/" + cfgFile)
+	} else {
+		cfgConfig.SetConfigFile(configDir + "/" + "config.yaml")
+	}
+
+	minSubmit = cfgConfig.GetInt("min_submit")
+	minCfm = cfgConfig.GetInt("min_confirmation")
+	CfmChainLength = cfgConfig.GetInt("confirmation_chain_length")
+
+	// ~/.zcn/network.yaml
+	cfgNetwork.AddConfigPath(configDir)
+	if &networkFile != nil && len(networkFile) > 0 {
+		cfgNetwork.SetConfigFile(configDir + "/" + networkFile)
+	} else {
+		cfgNetwork.SetConfigFile(configDir + "/" + "network.yaml")
+	}
+
+	if err := cfgConfig.ReadInConfig(); err != nil {
+		ExitWithError("Can't read config:", err)
+	}
+
+	cfgNetwork.ReadInConfig() //nolint: errcheck
+
 	// TODO: move the private key storage to the keychain or secure storage
-	var walletFilePath string
+	// ~/.zcn/wallet.json
 	if &walletFile != nil && len(walletFile) > 0 {
```

Comment:
it's impossible that the walletFile pointer is nil

Reply:
fixed